Library/Homebrew/cmd/install.rb

```diff
@@ -277,6 +277,15 @@ def install
         puts "To install one of them, run (for example):\n  brew install #{taps_search_results.first}"
       end
     end
+    return unless formulae.size > 1
```

Comment:
Will this handle caveats from dependencies also? Maybe we don't care about outputting all of them. Thoughts, @ilovezfs?

As a code comment: please turn this `unless` into an `if`.

Reply:
If we want to improve the signal-to-noise-ratio by re-printing caveats at the end only for formulae the user explicitly installed, it would be good to at least still do so for formulae that had originally been installed-on-request, even if they're merely being upgraded because they're outdated dependencies in this go.

Reply:
> Will this handle caveats from dependencies also?

currently, caveats for dependencies are not displayed. I'll look into this later

Reply:
@maxim-belkin Cool! As @ilovezfs mentioned I think `install_on_request` dependencies are a reasonable subset.